[BE] force DTensorTestBase.build_device_mesh to use world_size rather than NUM_DEVICES constant

XilunWu opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #116439

**Test**:
`python test/distributed/fsdp/test_shard_utils.py -k test_create_chunk_dtensor`


cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225

XilunWu committed:
[BE] force DTensorTestBase.build_device_mesh to use world_size rather than NUM_DEVICES constant

[ghstack-poisoned]

wanchaol reviewed (approved):
can you make this a separate PR instead of stacking this on the top of other layernorm PRs?

I think the current trunk is broken so we should land this quickly instead of landing together with other PRs

XilunWu committed:
Update on "[BE] force DTensorTestBase.build_device_mesh to use world_size rather than NUM_DEVICES constant"


**Test**:
`python test/distributed/fsdp/test_shard_utils.py -k test_create_chunk_dtensor`


cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu fegin wanchaol fduwjj wz337 tianyu-l wconstab yf225

[ghstack-poisoned]

XilunWu commented:
@pytorchbot merge
